frontend/src/routes/Governance/policies/CreatePolicy.tsx

```diff
@@ -88,16 +107,17 @@ export function CreatePolicy(props: { initialResources?: IResource[] }) {
           if (policy) {
             toast.addAlert({
               title: t('Policy created'),
-              message: t('{{name}} was successfully created.', { name: policy.metadata?.name }),
+              message: t('{{name}} was successfully created.', {
+                name: policy.metadata?.name,
+              }),
               type: 'success',
               autoClose: true,
             })
             submitForm()
-            history.push(
-              NavigationPath.policyDetails
-                .replace(':namespace', policy.metadata?.namespace ?? '')
-                .replace(':name', policy.metadata?.name ?? '')
-            )
+
+            return new Promise(() => {
+              setCreatedPolicy(policy)
```

Comment:
Where do you resolve the promise?

Reply:
Don't resolve it pass over to useeffect

Reply:
Wouldn't you want `PolicyWizard` to wait on this promise resolving to know when `onSubmit` has completed? It seems to me that after `setCreatedPolicy(policy)` is done calling, the promise should be resolved.

Reply:
onSubmit only accept promise. After the promise resolved, the running icon disappeared.. I wanted to control the running + disabled btn while watching policies updating. because of policies list, it causes "not found page bug". but Hard to observe policies updating in promise.

Reply:
Also the running + disabled btn should continue until history.push actually change the page (not execute)

Reply:
Actually... we can use @patternfly-labs/react-form-wizard 5.x.x it is more controllable but.... can we?

Reply:
now    ` "@patternfly-labs/react-form-wizard": "^1.28.0",`